pkg/cloud/provider.go

```diff
@@ -382,31 +383,31 @@ func ShareTenancyCosts(p Provider) bool {
 	return config.ShareTenancyCosts == "true"
 }
 
-func NewCrossClusterProvider(ctype string, overrideConfigPath string, cache clustercache.ClusterCache) (Provider, error) {
+func NewCrossClusterProvider(ctype string, config *config.ConfigFileManager, overrideConfigPath string, cache clustercache.ClusterCache) (Provider, error) {
```

Comment:
This is deprecated functionality, and I would love to have been the one to remove it, @Sean-Holcomb has a better idea of the places this touches and how to better document migration (if that's required). So, to make future removal less painful, I made some smaller API adjustments on my side to allow continued support.

Reply:
Instead of just providing a `ConfigFile` instance to the `ProviderConfig`, we now have to pass the `ConfigFileManager` as well. This is because the factory methods for the Provider include the `ConfigFileManager`. In order for a provider function to use `NewCrossClusterProvider` they need access to the `ConfigFileManager`, so we just carry that through the provider config and add a temporary `GetConfigFileManager()` method. All of this is required for this functionality that is supposedly deprecated. I made an attempt to remove it, but it runs fairly deep and was not an easy change. Once we can get @Sean-Holcomb to remove the NewCrossCluster support, then I can remove all the additional overhead required to support it.